packages/teleport/src/Discover/Kubernetes/LoginTrait/useLoginTrait.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect } from 'react';
+import useAttempt from 'shared/hooks/useAttemptNext';
+
+import TeleportContext from 'teleport/teleportContext';
+
+import type { User } from 'teleport/services/user';
+import type { AgentStepProps } from '../../types';
+import type { KubeMeta } from '../../useDiscover';
+
+export function useLoginTrait({ ctx, props }: Props) {
+  const [user, setUser] = useState<User>();
+  const { attempt, run, setAttempt, handleError } = useAttempt('processing');
+
+  const [dynamicUsers, setDynamicUsers] = useState<string[]>([]);
+  const [staticUsers, setStaticUsers] = useState<string[]>([]);
+
+  const [dynamicGroups, setDynamicGroups] = useState<string[]>([]);
+  const [staticGroups, setStaticGroups] = useState<string[]>([]);
```

Comment:
This states can be replaced with computation based on user.
```suggestion
const dynamicUsers = user?.traits.kubeUsers || [];
const dynamicGroups = user?.traits.kubeGroups || [];

// // TODO apply logic for kubes after this issue gets resolved:
// // https://github.com/gravitational/teleport/issues/17382
// const meta = props.agentMeta as KubeMeta;
// const staticUsers = meta.kube.users.filter(
//   user => !userDefinedKubeUsers.includes(user)
// );
// const staticGroups = meta.kube.groups.filter(
//   groups => !userDefinedKubeGroups.includes(groups)
// );
const staticUsers = [];
const staticGroups = [];
```

Reply:
wow your fast reviewer 😆 there was a big bug and i changed my data structure, though i think this comment still applies and i can save one re-render